base-notebook/container-entrypoint

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+# Add entries to /etc/passwd and /etc/group if being run as user ID
+# other than which the image has specified by the USER statement.
+
+fix-passwd-entry
```

Comment:
Is it possible for this logic to live in the `start*` scripts after tini runs instead of introducing additional scripts? I understand that Python and other programs may balk when the UID is not found in `/etc/users` and `/etc/group`, but does tini have that problem?

Simply looking to keep the number of separate workflow scripts we need to maintain low.

Reply:
The reason for adding it in an ``ENTRYPOINT`` script and doing it in there is that it gets applied even if someone overrides the ``CMD`` to do something else. For example, in JupyterHub, the ``CMD`` for the notebook images is overridden to execute ``jupyterhub-singleuser`` directly, bypassing ``start.sh``.

Reply:
This will depend on the Spawner. It is true of KubeSpawner, but isn't true of DockerSpawner in general (deployments *may* override CMD, but it is not overridden by default).

Doing this as an entrypoint makes sense to me. I think the same might be said for much of what's in start.sh, though. Effectively, putting this and only this in entrypoint is saying that the passwd entry is more important than and should be harder to opt-out of than anything in start.sh. That may be right, but I think we should think carefully about this in the future.